<launch>
  <arg name="robot_name"/>
  <arg name="master_width"        default="424"/>
  <arg name="master_height"        default="240"/>
  <arg name="align_depth"         default="true"/>
  <arg name="color_width"         default="$(arg master_width)"/>
  <arg name="color_height"        default="$(arg master_height)"/>
  <arg name="color_fps"         default="30"/>
  <arg name="depth_width"         default="$(arg master_width)"/>
  <arg name="depth_height"        default="$(arg master_height)"/>
  <arg name="depth_fps"         default="30"/>
  <arg name="infra_width"         default="$(arg master_width)"/>
  <arg name="infra_height"        default="$(arg master_height)"/>
  <arg name="infra_fps"         default="30"/>
  <arg name="enable_pointcloud"   default="true"/>
  <arg name="enable_infra1"   default="true"/>
  <arg name="enable_infra2"   default="true"/>
  <arg name="serial_no_camera1"    			default="049122250162"/> 			<!-- Note: Replace with actual serial number -->
  <arg name="serial_no_camera2"    			default="046322250258"/> 			<!-- Note: Replace with actual serial number -->
  <arg name="serial_no_camera3"    			default="838212072328"/> 			<!-- Note: Replace with actual serial number -->
  <arg name="serial_no_camera4"    			default="838212071329"/> 			<!-- Note: Replace with actual serial number -->
  <arg name="camera1"              			default="camera1"/>		<!-- Note: Replace with camera name -->
  <arg name="camera2"              			default="camera2"/>		<!-- Note: Replace with camera name -->
  <arg name="camera3"              			default="camera3"/>		<!-- Note: Replace with camera name -->
  <arg name="camera4"              			default="camera4"/>		<!-- Note: Replace with camera name -->
  <arg name="tf_prefix_camera1"         default="$(arg camera1)"/>
  <arg name="tf_prefix_camera2"         default="$(arg camera2)"/>
  <arg name="tf_prefix_camera3"         default="$(arg camera3)"/>
  <arg name="tf_prefix_camera4"         default="$(arg camera4)"/>
  <arg name="publish_tf"         default="true"/>
  <arg name="initial_reset"             default="true"/>

<!-- Hayashide D435i @Home -->
  <group ns="$(arg camera3)">
    <include file="$(find realsense2_camera)/launch/rs_camera.launch">
      <arg name="align_depth" value="$(arg align_depth)"/>
      <arg name="color_width" value="$(arg color_width)"/>
      <arg name="color_height" value="$(arg color_height)"/>
      <arg name="color_fps" value="$(arg color_fps)"/>
      <arg name="depth_width" value="$(arg depth_width)"/>
      <arg name="depth_height" value="$(arg depth_height)"/>
      <arg name="depth_fps" value="$(arg depth_fps)"/>
      <arg name="infra_width" value="$(arg infra_width)"/>
      <arg name="infra_height" value="$(arg infra_height)"/>
      <arg name="infra_fps" value="$(arg infra_fps)"/>
      <arg name="enable_pointcloud" value="$(arg enable_pointcloud)"/>
      <arg name="enable_infra1" value="$(arg enable_infra1)"/>
      <arg name="enable_infra2" value="$(arg enable_infra2)"/>
      <arg name="serial_no"             value="$(arg serial_no_camera3)"/>
      <arg name="tf_prefix"         		value="$(arg tf_prefix_camera3)"/>
      <arg name="publish_tf"         		value="$(arg publish_tf)"/>
      <arg name="enable_sync"         	value="true"/>
      <arg name="initial_reset"         value="$(arg initial_reset)"/>

      <arg name="filters"               value="pointcloud"/> <!-- resolve problem of watchdog freeze etc. -->

    </include>
  </group>
<!-- Motokura D435i -->
  <!-- <group ns="$(arg camera4)">
    <include file="$(find realsense2_camera)/launch/rs_camera.launch">
      <arg name="align_depth" value="$(arg align_depth)"/>
      <arg name="color_width" value="$(arg color_width)"/>
      <arg name="color_height" value="$(arg color_height)"/>
      <arg name="color_fps" value="$(arg color_fps)"/>
      <arg name="depth_width" value="$(arg depth_width)"/>
      <arg name="depth_height" value="$(arg depth_height)"/>
      <arg name="depth_fps" value="$(arg depth_fps)"/>
      <arg name="infra_width" value="$(arg infra_width)"/>
      <arg name="infra_height" value="$(arg infra_height)"/>
      <arg name="infra_fps" value="$(arg infra_fps)"/>
      <arg name="enable_pointcloud" value="$(arg enable_pointcloud)"/>
      <arg name="enable_infra1" value="$(arg enable_infra1)"/>
      <arg name="enable_infra2" value="$(arg enable_infra2)"/>
      <arg name="serial_no"             value="$(arg serial_no_camera4)"/>
      <arg name="tf_prefix"         		value="$(arg tf_prefix_camera4)"/>
      <arg name="publish_tf"         		value="$(arg publish_tf)"/>
      <arg name="enable_sync"         	value="true"/>
    </include>
  </group> -->
</launch>